src/vs/platform/terminal/node/terminalEnvironment.ts

```diff
@@ -87,8 +87,17 @@ export function getShellIntegrationInjection(
 	if (options.shellIntegration.nonce) {
 		envMixin['VSCODE_NONCE'] = options.shellIntegration.nonce;
 	}
+
+	const enableWindowsEnvReporting = options.windowsUseConptyDll || options.windowsEnableConpty && getWindowsBuildNumber() >= 22631;
```

Comment:
Let's stick this inside the isWindows block to avoid unnecessary work

Reply:
Just like [this](https://github.com/microsoft/vscode/pull/240710/commits/39f537068fb3eb66e93d5fd11cd8213d280c2853), right?